webui/src/data/server.js

```diff
@@ -530,6 +531,15 @@ class ServerCache {
         return file;
     }
 
+    getVersions(recordID, callback){
```

Comment:
In general we should prefer to not use callbacks, but to place the getter results in the store, which automatically triggers a repaint of the UI.

Reply:
I see, I noticed that we follow that pattern but I didn't know the difference. I will change it and create a versions store. Also, we already store the versions in the recordCache, but I couldn't use the record here because if serverCache returns an error when getting a record, the versions attribute is not accessible.

Reply:
I think I would change the callback here: https://github.com/EUDAT-B2SHARE/b2share/blob/master/webui/src/data/server.js#L435 to test that the return code is 410 (deleted record) and if so, retrieve the versions and place them right in the 'Error' object of that record. You're right we don't have a direct link to versions anymore, which is a pity.

Reply:
Ah yes much better, I will change it as described